WordPress/Classes/ViewRelated/Blog/BlogSelectorViewController.m

```diff
@@ -186,6 +186,15 @@ - (IBAction)cancelButtonTapped:(id)sender
     }
     
     if (self.dismissOnCancellation) {
+        [self dismissViewController];
+    }
+}
+
+- (void)dismissViewController
+{
+    if (self.isModal && self.isRootInNavigation) {
+        [self dismissViewControllerAnimated:YES completion:nil];
+    } else {
```

Comment:
During development, it was being pushed into a UINavigationController instance. At some point i switched it all to modal, so indeed, i guess we can just assume it'll always be that way. For now, at least.

Updating, thanks!